packages/codegen-ui/lib/utils/form-component-metadata.ts

```diff
@@ -93,7 +92,11 @@ export const mapFormMetadata = (form: StudioForm, formDefinition: FormDefinition
           });
         }
 
-        if ((element.relationship || isEnumFieldType(element.dataType)) && 'valueMappings' in element) {
+        if (
+          (element.relationship ||
+            (element.dataType && typeof element.dataType === 'object' && 'enum' in element.dataType)) &&
```

Comment:
can this reuse `isEnumDataType` in render-checker.ts?

Reply:
This code has been removed